include/cglm/euler.h

```diff
@@ -448,4 +454,168 @@ glm_euler_by_order(vec3 angles, glm_euler_seq ord, mat4 dest) {
   dest[3][3] = 1.0f;
 }
 
+
+/*!
+ * @brief creates NEW quaternion using rotation angles and does
+ *        rotations in x y z order (roll pitch yaw)
+ * 
+ * @param[out]  q     quaternion
+ * @param[in]   angle angles x y z (radians)
+ */
+CGLM_INLINE
+void
+glm_euler_xyz_quat(versor q, vec3 angles) {
+  float xs = sinf(angles[0] / 2.0f);
+  float xc = cosf(angles[0] / 2.0f);
+
+  float ys = sinf(angles[1] / 2.0f);
+  float yc = cosf(angles[1] / 2.0f);
+  
+  float zs = sinf(angles[2] / 2.0f);
+  float zc = cosf(angles[2] / 2.0f);
```

Comment:
I know new compiler and C versions can handle this but I prefer to do declarations beginning of a scope block:

```C
  float xs, xc, ys, yc, zs, zc;

  xs = sinf(angles[0] * 0.5f);
```

also even compilers do optimizations it would be nice to multiply with 0.5 rather than dividing by 2.